submissions/akulahere/task-js-post-oop/index.js

```diff
@@ -0,0 +1,76 @@
+class Inhabitant {
+  constructor(species, name, gender, legs, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = 'No friends' ;
+  }
+
+  makeFriend(friend) {
+    this.friends = this.friends === 'No friends' ? [] : this.friends;
+    this.friends.push(friend.name);
+  }
+  getProperties() {
+    return ['species', 'name', 'gender', 'legs', 'saying', 'friends'];
+  }
+
+  makeMessage() {
+    return this.getProperties().map(property => `${capitalizeString(property)}: ${this[property]}`).join(' ')
+  }
+}
+
+class Dog extends Inhabitant {
+  constructor(name, gender) {
+    super('dog', name, gender, 4, 'woof-woof!');
+  }
+}
+
+class Cat extends Inhabitant {
+  constructor(name, gender) {
+    super('cat', name, gender, 4, 'mrrr!');
+  }
+}
+
+class CatWoman extends Cat {
+  constructor(name) {
+    super(name, 'female');
+    this.species = 'CatWoman';
+    this.legs = 2;
+    this.hands = 2;
+  }
+  getProperties() {
+    let properties = super.getProperties();
+    properties.splice(4, 0, 'hands');
```

Comment:
What will happen if a peer developer adds property `eyes` in `Inhabitants` at an arbitrary position?
You may want to add `hands` somewhere next to `legs` (if my assumption is coreect). JS offers instruments to make this fragment of code more obvious. Seek among `Array` methods.

Reply:
Replaced exact index with indexOf method

Reply:
Now hands always insert before legs.